### DISCUSSION TITLE
Am I correct that I can only import 20 Issues at a time?

### DISCUSSION THREAD

**Discussion by joachimweyl** [gavinr/github-csv-tools#80] [Q&A] [answered]
Every time I try to import more than 20 issues I receive this error "You have exceeded a secondary rate limit and have been temporarily blocked from content creation. Please retry your request again later." I am confused by this because I thought the rate limit was 5000 calls per hour which should allow more than 20 issues. I also think that the remaining limit I get of 4869 means I should not have hit the limit yet but obviously, I am missing something.
not sure if this helps
```
      'x-ratelimit-limit': '5000',
      'x-ratelimit-remaining': '4869',
      'x-ratelimit-reset': '1660682593',
      'x-ratelimit-resource': 'core',
      'x-ratelimit-used': '131',
```
I would like to use this tool to move a little under 900 issues and if I have to break this into batches of 20 each hour that is going to take a while.

**Comment by gavinr** [comment]
I'm not quite sure, but this PR is related to rate limits: https://github.com/gavinr/github-csv-tools/pull/78 - can you test out the code on that PR and see if it resolves your issue?

**Comment by joachimweyl** [comment]
I tested it and still getting "RequestError [HttpError]: You have exceeded a secondary rate limit and have been temporarily blocked from content creation. Please retry your request again later."
20 issues imported again.
```
      'x-ratelimit-limit': '5000',
      'x-ratelimit-remaining': '4979',
      'x-ratelimit-reset': '1660700644',
      'x-ratelimit-resource': 'core',
      'x-ratelimit-used': '21',
```

**Comment by joachimweyl** [comment]
Updated to the newest version with PR #82 (as #78 was closed in favor of 82). Getting the following error.
```
[@octokit/plugin-throttling] `onAbuseLimit()` is deprecated and will be removed in a future release of `@octokit/plugin-throttling`, please use the `onSecondaryRateLimit` handler instead
Abuse detected for request POST /repos/{owner}/{repo}/issues
Error
RequestError [HttpError]: You have exceeded a secondary rate limit and have been temporarily blocked from content creation. Please retry your request again later.
```
Once again this happened after 20 issues were imported. I tested with my personal csv and with `test/5.csv` they both failed after 20 issues were imported.

**Reply by gavinr** [reply]
hm, I tested it and the new update allowed me to create 49 issues without hitting any limits. Is it possible you're on a shared public IP where others are using the GitHub API and that is playing into the limits?

**Reply by joachimweyl** [reply]
Personal Comcast IP, not public. I deleted my local install just in case and cloned it again. Waited well past ratelimit-reset. Tried again. I am still getting the same error. Notice it has only used 28 of the 5000 available. Not sure why it is failing.
```
      'x-ratelimit-limit': '5000',
      'x-ratelimit-remaining': '4972',
      'x-ratelimit-reset': '1660750577',
      'x-ratelimit-resource': 'core',
      'x-ratelimit-used': '28'
```
not sure if this is helpful but figured I could share it in case.
```
    at C:\Users\joach\AppData\Roaming\npm\node_modules\github-csv-tools\node_modules\@octokit\request\dist-node\index.js:86:21
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async Job.doExecute (C:\Users\joach\AppData\Roaming\npm\node_modules\github-csv-tools\node_modules\bottleneck\light.js:405:18)
```

**Reply by joachimweyl** [reply]
I forgot I had installed it with the installer. After updating with the installer instead of just my local repo I was able to import test/5.csv. Apologies for the confusion.

**Comment by joachimweyl** [comment, marked as answer]
The newest version works. Thank you!
